packages/react-native-reanimated/Common/cpp/reanimated/LayoutAnimations/LayoutAnimationsProxy_Experimental.cpp

```diff
@@ -0,0 +1,850 @@
+}
+
+bool LayoutAnimationsProxy_Experimental::shouldOverridePullTransaction() const {
+  return true;
+}
+
+// MARK: Light Tree
+
+void LayoutAnimationsProxy_Experimental::updateLightTree(
+    const PropsParserContext &propsParserContext,
+    const ShadowViewMutationList &mutations,
+    ShadowViewMutationList &filteredMutations) const {
+  ReanimatedSystraceSection s("updateLightTree");
+  std::unordered_set<Tag> moved, deleted;
+  for (auto it = mutations.rbegin(); it != mutations.rend(); it++) {
+    const auto &mutation = *it;
+    switch (mutation.type) {
+      case ShadowViewMutation::Delete: {
+        deleted.insert(mutation.oldChildShadowView.tag);
+        break;
+      }
+      case ShadowViewMutation::Insert: {
+        moved.insert(mutation.newChildShadowView.tag);
+        break;
+      }
+      case ShadowViewMutation::Remove: {
+        const auto tag = mutation.oldChildShadowView.tag;
+        if (deleted.contains(tag)) {
+          lightNodes_[tag]->intent = TO_DELETE;
+        } else if (moved.contains(tag)) {
+          lightNodes_[tag]->intent = TO_MOVE;
+        }
+        break;
+      }
+      default: {
+      }
+    }
+  }
+
+  for (const auto &mutation : mutations) {
+    maybeUpdateWindowDimensions(mutation);
+    switch (mutation.type) {
+      case ShadowViewMutation::Update: {
+        auto &node = lightNodes_[mutation.newChildShadowView.tag];
+        if (!node) {
+          node = std::make_shared<LightNode>();
+        }
+        node->previous = mutation.oldChildShadowView;
+#ifdef ANDROID
+        if (node->current.props) {
+          // on android rawProps are used to store the diffed props
+          // so we need to merge them
+          // this should soon be replaced in RN with Props 2.0 (the diffing will
+          // be done at the end of the pipeline)
+          auto &currentRawProps = node->current.props->rawProps;
+          auto mergedRawProps = folly::dynamic::merge(currentRawProps, mutation.newChildShadowView.props->rawProps);
+          node->current = mutation.newChildShadowView;
+          node->current.props =
+              getComponentDescriptorForShadowView(node->current)
+                  .cloneProps(propsParserContext, mutation.newChildShadowView.props, RawProps(mergedRawProps));
+        } else {
+          node->current = mutation.newChildShadowView;
+        }
+#else
+        node->current = mutation.newChildShadowView;
+#endif
+        auto tag = mutation.newChildShadowView.tag;
+        if (layoutAnimationsManager_->hasLayoutAnimation(tag, LAYOUT)) {
+          layout_.push_back(node);
+        } else {
+          filteredMutations.push_back(mutation);
+        }
+        break;
+      }
+      case ShadowViewMutation::Create: {
+        auto &node = lightNodes_[mutation.newChildShadowView.tag];
+        node = std::make_shared<LightNode>();
+        node->current = mutation.newChildShadowView;
```

Comment:
How about this instead?
```suggestion
        const auto &node = std::make_shared<LightNode>();
        node->current = mutation.newChildShadowView;
        lightNodes_[mutation.newChildShadowView.tag] = node;
```